.github/workflows/rollback.yml

```diff
@@ -221,14 +244,15 @@ jobs:
           aws-region: us-west-2
       - name: Replace latest with rollback version
         run: |
-          VERSION_KEY="${{ matrix.os }}/${{ matrix.arch }}/${{ github.event.inputs.rollback_version }}/aws-otel-collector.${{ matrix.type }}"
+          VERSION_KEY="${{ matrix.os }}/${{ matrix.arch }}/${{ github.event.inputs.rollback-version }}/aws-otel-collector.${{ matrix.type }}"
           LATEST_KEY="${{ matrix.os }}/${{ matrix.arch }}/latest/aws-otel-collector.${{ matrix.type }}"
-          VERISON_URI="s3://$S3_BUCKET/$VERSION_KEY"
-          LATEST_URI="s3://$S3_BUCKET/$LATEST_KEY"
-          aws s3 cp $VERSION_URI $LATEST_URI --acl public-read
+          VERISON_URI="s3://${S3_BUCKET}/${VERSION_KEY}"
+          LATEST_URI="s3://${S3_BUCKET}/${LATEST_KEY}"
+          aws s3 cp "${VERSION_URI}" "${LATEST_URI}" --acl public-read
   ssm-rollback:
     runs-on: ubuntu-latest
     needs: [all-checked]
+    if: always() && needs.all-checked.result == 'success' && github.event.inputs.skip-ssm-rollback == 'false'
```

Comment:
Wait why would we want to use always with needs.all-checked.result though?

Reply:
It doesn't work otherwise. It seems like if a job is skipped, every dependent job is also skipped. So we need to get around this by having the always with the check to basically have the same effect as the normal needs.

Reply:
Weird to me with the Github conditions. Cause it only needs the ```all-checked``` job and does not have any children job.

Reply:
I've tested this in a fork. Without the condition, it won't even run this job.